### DISCUSSION TITLE
JS APIs: What is the boundary between this and SoJS?

### DISCUSSION THREAD

**Discussion by LeaVerou** [Devographics/surveys#118] [State of HTML Design]
State of JS has a [Browser APIs](https://2022.stateofjs.com/en-US/features/browser-apis/) section, which asks about many browser-specific JS APIs.

There is generally agreement that we would need to ask about *some* JS APIs here, but where is the boundary and how much overlap do we want to have (if any)?
Relevant, @chrishtr, listed several JS APIs in his [vision about State of HTML](https://github.com/Devographics/surveys/discussions/111#discussioncomment-6500417):
* Missing DOM APIs
* Events & observation patterns
* Editing (contentEditable and newer specs around it?)
* Selection (Range APIs?)
* Template rendering and state management (Template parts?)
* JS extensions to the rendering pipeline

**Comment by captainbrosset** [comment]
Given the name of this survey (State of HTML), I think there's value in asking about JS APIs without worrying too much about the overlap with the SoJS survey. My thought is that the people who take SoHTML are not going to be exactly the same than the ones who take SoJS. They may self identify as people who mostly do HTML, and little JS, even if they do use JS every once in a while. And we do want to hear from them too!

**Comment by mfreed7** [comment]
I think the State of JS is more about the language itself - things specified by TC39. The list of topics in the OP are all things specified in WHATWG/html and WHATWG/dom. I think that makes the distinction pretty clean, and I don't think there'll be too much overlap. If there's a bit of overlap, that's probably ok too. As pointed out above, it'll be interesting to see how things are prioritized by the folks who take the SoHTML survey.

**Reply by LeaVerou** [reply]
> I think the State of JS is more about the language itself - things specified by TC39. 

From the OP: 

> State of JS has a [Browser APIs](https://2022.stateofjs.com/en-US/features/browser-apis/) section, which asks about many browser-specific JS APIs.

That said, maybe moving towards asking primarily about ECMAScript features is a good future direction for State of JS?

**Comment by SachaG** [comment]
Like others have pointed out the State of JS tends to focus more on A) the language itself and B) JavaScript frameworks. So I think it would make sense for the State of HTML survey to become the place where we ask about browser APIs and other things that the typical React developer (e.g. me) might not interact with on a daily basis anyway.
